### ISSUE TITLE
Why not do scheduled updates in dns_resolver.go?

### ISSUE THREAD

**Issue by hankji** [grpc/grpc-go#6604]
In the func watcher of dns_resolver.go:
`
func (d *dnsResolver) watcher() {
	...
		if err == nil {
			// Success resolving, wait for the next ResolveNow. However, also wait 30
			 seconds at the very least to prevent constantly re-resolving.
			backoffIndex = 1
			timer = newTimerDNSResRate(minDNSResRate)
			select {
			case <-d.ctx.Done():
				timer. Stop()
				return
			case <-d.rn:
			}
		} else {
	...
}
`

why block it until the resolveNow was been called? (in my case, if there add some new server pod, it will not be receive request.)
Is there a problem if adding a timer?
`
        select {
	case <-d.ctx.Done():
		timer. Stop()
		return
        case <-timer.C:
	case <-d.rn:
	}

`
thank u very much!

**Comment by huangchong94** [comment]
I think there are reasons

To protect dns sever from too much traffic. 

To prevent client from doing too much unnecessary query,since most of the time the results will be the same

**Comment by easwars** [comment]
Please read through this issue for more context: https://github.com/grpc/grpc/issues/12295

TL;dr
- Sorry, gRPC's implementation of the DNS resolver will not support a configurable timeout for re-resolution
- You are free to implement your own custom resolver though, that can do this
- You could also use the `max_connection_age` feature to ensure that the server drops connections after they reach the `max_connection_age`, forcing the client to re-resolve. See: https://pkg.go.dev/google.golang.org/grpc/keepalive#ServerParameters

Thanks for your inquiry. Feel free to open a new issue if you still have questions/concerns after reading the issue mentioned at the top of this comment.
